### PR TITLE
Add filters and merged into one blade file

### PR DESCRIPTION
**Pull request by buzkall**
Add filters using a Trait and merge all the blade files into a single one

208239423-3251f7bc-aeba-44d4-8f81-e04924987945

### CODE REVIEW

File: src/Widgets/SourcesWidget.php [hunk 1]
```diff
@@ -2,21 +2,26 @@
...
+    public static string $tableHeader = 'Page';
+    public string $field = 'page';
+    public string $filter = 'today';
 
     protected function getViewData(): array
     {
         return [
             'items' => PageView::query()
-                ->scopes(['filter' => [$this->period]])
+                               ->scopes(['filter' => [$this->filter]])
```
Comment (andreaselia) [inline src/Widgets/SourcesWidget.php]: Indentation fix, something like the below to match the other lines.
```suggestion
                ->scopes(['filter' => [$this->filter]])
```

File: resources/views/widgets/widget.blade.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                @endif
+
+                @if ($filters)
+                    <select
+                            wire:model="filter"
+                            @class([
+                                'text-gray-900 border-gray-300 block h-10 transition duration-75 rounded-lg shadow-sm focus:border-primary-500 focus:ring-1 focus:ring-inset focus:ring-primary-500',
+                                'dark:bg-gray-700 dark:border-gray-600 dark:text-gray-200 dark:focus:border-primary-500' => config('filament.dark_mode'),
+                            ])
+                            wire:loading.class="animate-pulse"
```
Comment (andreaselia) [inline resources/views/widgets/widget.blade.php]: Could we indent this back a tab, please? Everything should indent 4 spaces, and this looks like 8?

**Review by andreaselia — CHANGES_REQUESTED** [review]
Thank you very much for your PR, it is greatly appreciated. I've made a couple of comments around formatting, but the rest looks great. Could you also update your initial PR message with a screenshot of the works (visual) for future reference too, please?

**Comment by buzkall** [comment]
Done!
I've been doing test on my local installation and I've done another version with just one widget and 4 cards

Something like this:

```
protected function getCards(): array
{
    return [
        Card::make(
            __('Devices'),
            ['items' => PageView::query()
                                ->scopes(['filter' => [$this->filter]])
                                ->select('device as type', DB::raw('count(*) as users'))
                                ->groupBy('type')
                                ->get(),
             'field' => 'type',
             'heading' => [__('Type'), __('Users')],
            ]
        ),

        Card::make(
            __('Users'),
            ['items' => PageView::query()
                                ->scopes(['filter' => [$this->filter]])
                                ->where('device', '!=', 'robot')
                                ->select('country', DB::raw('count(*) as users'))
                                ->groupBy('country')
                                ->get(),
             'field' => 'country',
             'heading' => [__('Country'), __('Users')],
            ]
        ),

        Card::make(
            __('Pages'),
            ['items' => PageView::query()
                                ->scopes(['filter' => [$this->filter]])
                                ->where('device', '!=', 'robot')
                                ->select('uri as page', DB::raw('count(*) as users'))
                                ->groupBy('page')
                                ->get(),
             'field' => 'page',
             'heading' => [__('Page'), __('Users')],
            ]
        ),

        Card::make(
            __('Sources'),
            ['items' => PageView::query()
                                ->scopes(['filter' => [$this->filter]])
                                ->where('device', '!=', 'robot')
                                ->select('source as page', DB::raw('count(*) as users'))
                                ->whereNotNull('source')
                                ->groupBy('source')
                                ->get(),
             'field' => 'page',
             'heading' => [__('Page'), __('Users')],
            ]
        ),
    ];
}
```

with a foreach in the view to get the cards

```
@foreach ($this->getCachedCards() as $card)
```

This way a change in a filter affects the other three but doesn't affect any other filter of another dashboard widget

**Comment by andreaselia** [comment]
I don't have much knowledge of Filament (still quite new to it), would you recommend we do it that way in the package or keep it as-is?

**Comment by buzkall** [comment]
I feel it's more clear with just one widget. I've uploaded two new files for you to check it: the widget with the four cards and the view. I've also changed some classes in the view, because when using the dark mode in filament, the data can't be read

**Comment by andreaselia** [comment]
Thank you for your contribution @buzkall. Hopefully I haven't broken anything from my small code style amendments, but I've merged this in now and pushed a release as v1.1.0.

**Pull request merged**